qa/os/src/test/java/org/elasticsearch/packaging/test/PackageTests.java

```diff
@@ -298,7 +299,23 @@ public void test81CustomPathConfAndJvmOptions() throws Exception {
         });
     }
 
-    public void test82SystemdMask() throws Exception {
+    public void test82CustomJvmOptionsDirectoryFile() throws Exception {
+        final Path heapOptions = installation.config(Paths.get("jvm.options.d", "heap.options"));
+        try {
+            append(heapOptions, "-Xms512m\n-Xmx512m\n");
+
+            startElasticsearch();
+
+            final String nodesResponse = makeRequest(Request.Get("http://localhost:9200/_nodes"));
+            assertThat(nodesResponse, CoreMatchers.containsString("\"heap_init_in_bytes\":536870912"));
```

Comment:
Nit - I'd use a static import of `Matchers`, personally.

Reply:
I pushed 3a2bf7bb82895991d8cb0c66304938bcccc47b72.